odoo/addons/openupgrade_records/models/openupgrade_record.py

```diff
@@ -111,3 +114,40 @@ def field_dump(self):
                 dict([(x.name, x.value) for x in record.attribute_ids]))
             data.append(repre)
         return data
+
+    @api.model
+    def list_modules(self):
+        """ Return the set of covered modules """
+        self.env.cr.execute(
+            """SELECT DISTINCT(module) FROM openupgrade_record
+            ORDER BY module""")
+        return [module for module, in self.env.cr.fetchall()]
+
+    @staticmethod
+    def _read_manifest(addon_dir):
+        for manifest_name in MANIFEST_NAMES:
+            if os.access(os.path.join(addon_dir, manifest_name), os.R_OK):
+                with open(os.path.join(addon_dir, manifest_name), 'r') as f:
+                    manifest_string = f.read()
+                    return ast.literal_eval(manifest_string)
+        raise ValidationError('No manifest found in %s' % addon_dir)
```

Comment:
Why not simply `return {}` for ignoring invalid folders? At the end, they are harmless for this purpose, and we avoid possible crashes.

Reply:
Stefan, what do you think about this?

Reply:
Sorry for the delay! About your question, the method is only called on the list of modules that are returned by the method `list_modules` above (see https://github.com/odoo/odoo/commit/3d34d5838#diff-cadd7a2530955f21c93bbe4372640d0bbb4afce512c9a2b0b6f0dbbc96585096R83), so this is a classic case of "should not happen". As such, I think we can leave it like this.

Reply:
OK